app/scripts/vendor/coseme-client/client.js

```diff
@@ -366,42 +383,56 @@
       function onMessageReceived(messageId, senderJID, content,
                                  timestamp, wantsReceipt, senderName,
                                  isBroadcast, authorJID) {
-        fire('message', pub,
-             newTextMessage(messageId, senderJID, content,
-                            senderName, authorJID));
+        fire('message', pub, newTextMessage(
+          messageId,
+          senderJID,
+          content,
+          timestamp,
+          senderName,
+          authorJID
+        ));
         methods.call('message_ack', [senderJID, messageId]);
         // TODO: The pub should be the client, we need to refactor this
         // module in order to provide the `singleton` module in a smarter way.
       }
 
       function onAudioReceived(messageId, senderJID, url, size,
-                               receiptRequested, isBroadcast) {
+                               receiptRequested, isBroadcast,
+                               pushName, timestamp) {
 
         onMultiMediaReceived('audio', messageId, senderJID, undefined, url,
-                             size, receiptRequested, isBroadcast);
+                             size, receiptRequested, isBroadcast,
+                             pushName, timestamp);
       }
 
       function onGroupAudioReceived(messageId, groupJID, authorJID, url, size,
-                                    receiptRequested) {
+                                    receiptRequested, pushName, timestamp) {
 
         onMultiMediaReceived('audio', messageId, groupJID, undefined, url,
-                             size, receiptRequested, false, authorJID);
+                             size, receiptRequested, false,
+                             authorJID, timestamp);
       }
 
       function onGroupMultiMediaReceived(type, messageId, groupJID, authorJID,
                                          preview, url, size, receiptRequested,
-                                         isBroadcast) {
```

Comment:
I removed it to get it in-line with coseme. Please look at `app/components/coseme/src/yowsup_if.js`, line 384 and the following:

```javascript
      var mediaProcessor = {
        // These functions are surprisingly similar, aren't they?...
        image: function(childNode) {
          if (isGroup) {
            _signalInterface.send("group_imageReceived",
                                  [msgId, fromAttribute, author, mediaPreview,
                                  mediaUrl, mediaSize, wantsReceipt, pushName,
                                  timestamp]);
```
Coseme does not send broadcast-request information, but pushName as 9th argument. This is why I've changed it. Or do I miss something?